tests/unit/reaction/test_combinatorics.py

```diff
@@ -206,3 +213,49 @@ def test_generate_permutations(
     assert spin_permutations[1][2][1] == +1
     assert spin_permutations[2][0][1] == +1
     assert spin_permutations[3][0][1] == +1
+
+
+def test_perform_external_edge_identical_particle_combinatorics(
+    particle_database: ParticleCollection,
+):
+    double_decay = Topology(
+        nodes={0, 1, 2},  # type: ignore
+        edges={
+            0: Edge(originating_node_id=None, ending_node_id=0),
+            1: Edge(originating_node_id=0, ending_node_id=1),
+            2: Edge(originating_node_id=0, ending_node_id=2),
+            3: Edge(originating_node_id=1, ending_node_id=None),
+            4: Edge(originating_node_id=1, ending_node_id=None),
+            5: Edge(originating_node_id=2, ending_node_id=None),
+            6: Edge(originating_node_id=2, ending_node_id=None),
+        },
+    )
+    graph = StateTransitionGraph[ParticleWithSpin](
+        topology=double_decay,
+        edge_props={
+            0: (particle_database["J/psi(1S)"], 0),
+            1: (particle_database["f(0)(980)"], 0),
+            2: (particle_database["f(0)(980)"], 0),
+            3: (particle_database["pi-"], 0),
+            4: (particle_database["pi+"], 0),
+            5: (particle_database["pi-"], 0),
+            6: (particle_database["pi+"], 0),
+        },
+        node_props={},
+    )
+    graphs = perform_external_edge_identical_particle_combinatorics(graph)
+    assert len(graphs) == 4
+
+    final_state_edge_ids = graph.topology.outgoing_edge_ids
+    get_final_state = lambda g: tuple(  # noqa: E731
+        g.get_edge_props(i)[0].name for i in final_state_edge_ids
+    )
+    get_originating_node = lambda g: tuple(  # noqa: E731
+        g.topology.edges[i].originating_node_id for i in final_state_edge_ids
+    )
+    for g in graphs:  # pylint: disable=invalid-name
+        assert get_final_state(g) == ("pi-", "pi+", "pi-", "pi+")
+    assert get_originating_node(graphs[0]) == (1, 1, 2, 2)
+    assert get_originating_node(graphs[1]) == (1, 2, 2, 1)
+    assert get_originating_node(graphs[2]) == (2, 1, 1, 2)
+    assert get_originating_node(graphs[3]) == (2, 2, 1, 1)
```

Comment:
This is quite interesting actually. In this example (using this topology) there are actually only 3 valid final state combinations! So we have to clearly specify what this code is supposed to do. So is this function supposed to generate all combinations, or only the physically meaningful? I have to dig a bit deeper here as well

Reply:
Indeed, seems it comes down to what we want the `reaction` module to return as a whole. What is the meaning of 'solutions'?

Reply:
Solution currently means one completely specific amplitude transition. So changing the slightest bit of info which is physically not the identical, would result in a different solution.

So to me `perform_external_edge_identical_particle_combinatorics` suggests it will only create the physically relevant combinations. Also the code doc string explicitly states that only identical particles originating from different nodes can be swapped. So only the physically relevant combinations is def the responsibility of this function.

What I did not realize just now is that it also matters what kind of state is assigned to the intermediate edges... So actually this result would actually be correct. And we can do this combinatorics generally, even if the two intermediate state would be identical in this example. Then theoretically swapping the two final state groups completely (11, 22) vs (22, 11) are actually the same solution, but with the correct normalization (1/2 on amp level), `sympy.simplify` can combine these two again.

So all in all, i would say the code is fine and doing what it should.